torch_ipex/csrc/cpu/ExtendOPs.cpp

```diff
@@ -305,11 +305,11 @@ inline std::vector<at::Tensor> _interaction_backward(const at::Tensor & grad_out
 
 at::Tensor AtenIpexTypeExt::interaction_forward(const std::vector<at::Tensor> & input) {
   if (input[0].scalar_type() == at::kFloat) {
-    for (const auto &in : input) { TORCH_INTERNAL_ASSERT(in.scalar_type() == at::kFloat); }
```

Comment:
@hongzhen1 , Do you think it should be replaced by TORCH_CHECK?

Reply:
it's OK for me